tests/util_test.py

```diff
@@ -141,3 +141,40 @@ def simple_name_marker(contact, name, soup):
         assert out == util.autolink(
             inp, person_processor=simple_name_marker,
             url_processor=None)
+
+
+def test_parsing_hashtags():
```

Comment:
thanks so much for adding test cases! could you add a couple more :)

1. # inside a larger string
2. inside a URL (e.g. `http://example.com/path#fragment`)